x/staging/posts/types/report.go

```diff
@@ -8,27 +8,48 @@ import (
 )
 
 // NewReport returns a Report
-func NewReport(postID string, reportType string, message string, user string) Report {
+func NewReport(postID string, reasons []string, message, user string) Report {
 	return Report{
 		PostID:  postID,
-		Type:    reportType,
+		Reasons: reasons,
 		Message: message,
 		User:    user,
 	}
 }
 
+// AreReasonsValid checks if the report reasons are present inside the paramsReasons
+func (r Report) AreReasonsValid(paramsReasons []string) bool {
+	exists := make(map[string]bool, len(paramsReasons))
+	for _, reason := range paramsReasons {
+		exists[reason] = true
+	}
+
+	for _, rr := range r.Reasons {
+		if !exists[rr] {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Probably this can be moved as a keeper method instead? Something like a `k.ValidateReport` method similar to `k.ValidateProfile`?

Reply:
There's already the `CheckReportValidity` keeper method that pack this method and the `Validate()` one.
I think that should be enough, otherwise we should move both in keeper.

Reply:
The I think it's better to move this logic inside the `CheckReportValidity` method itself instead of having it here. This would make sure that we do not need to remember to call two different methods to validate a report.